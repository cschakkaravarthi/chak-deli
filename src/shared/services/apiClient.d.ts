declare const client: import("axios").AxiosInstance;
export default client;
//# sourceMappingURL=apiClient.d.ts.map