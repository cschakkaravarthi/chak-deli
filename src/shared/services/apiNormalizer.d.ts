import Article from '../models/Article.model';
import Bookmark from '../models/Bookmark.model';
import BoxModel from '../models/Box.model';
import EventModel from '../models/Event.model';
import Link, { LinkGroupItems } from './../models/Link.model';
import Notification from '../models/Notification.model';
import OfficesPeople, { OfficeItem } from './../models/OfficesPeople.model';
import PeopleModel from '../models/People.model';
import Todo from '../models/Todo.model';
import { CatalogSearch } from '../models/Catalog.model';
import { ContactGroup, Department } from '../models/Department.model';
import { DocumentSearch } from '../models/Document.model';
import { KnowledgeSearch } from '../models/KnowledgeBase.model';
import { Manager, Users } from '../models/OrgChart.model';
import { Office, OfficeSearch } from '../models/Office.model';
import { Page, PageSearch } from '../models/Page.model';
import { ResponseData, SuccessfulResponse } from './apiService';
import { Categories, ManualNotification, ManualNotificationListModel, ManualNotifications, NotificationBell } from '../models/ManualNotification.model';
import UserInfoModel, { UserProfileModel, UserProfilePicture, ContactList, UserProfileUpdateResponse, ContactResponse } from '../../shared/models/UserInfo.model';
import { TaxonomyTermModel, VariousContentGroupModel, VariousContentModel } from '../types/contentTypes';
export default class ApiNormalizer {
    /**
     * Sanitation and validation of API response is crucial to prevent bugs and XSS.
     */
    static normalizeContent(response: ResponseData): VariousContentModel[];
    static normalizeGroupedContent(response: ResponseData): VariousContentGroupModel[];
    static normalizeContentSetGroup(response: ResponseData): VariousContentGroupModel | null;
    static normalizeContentSet(response: ResponseData): VariousContentModel | null;
    static validateArticle(content: ResponseData): ResponseData | null;
    static validateEvent(content: ResponseData): ResponseData | null;
    static validateDepartment(content: ResponseData): ResponseData | null;
    static validateOffice(content: ResponseData): ResponseData | null;
    static validatePeople(content: ResponseData): ResponseData | null;
    static validateNotification(content: ResponseData): ResponseData | null;
    static validateLinks(content: ResponseData): ResponseData | null;
    static validateBox(content: ResponseData): ResponseData | null;
    static validateSuccessfulResponse(content: ResponseData): ResponseData | null;
    static normalizeMinisiteLink(untrustedContent: ResponseData): Link | null;
    static normalizeSuccessfulResponse(untrustedContent: ResponseData): SuccessfulResponse | null;
    static normalizeBox(untrustedContent: ResponseData): BoxModel | null;
    /**
     * Convert API response into the data model we want to store in state.
     */
    static normalizeArticle(untrustedContent: ResponseData): Article | null;
    static normalizeTaxonomyTerm(content: ResponseData): TaxonomyTermModel | null;
    static normalizeTaxonomyTerms(content: ResponseData): TaxonomyTermModel[];
    static normalizeEvent(untrustedContent: ResponseData): EventModel | null;
    static normalizeGroupedContacts(contactsString: string): ContactGroup[];
    static normalizeDepartment(untrustedContent: ResponseData): Department | null;
    static normalizePerson(untrustedContent: ResponseData): PeopleModel | null;
    static validatePageSearch(content: ResponseData): ResponseData | null;
    static normalizePageSearch(untrustedContent: ResponseData): PageSearch | null;
    static validateOfficeSearch(content: ResponseData): ResponseData | null;
    static normalizeOfficeSearch(untrustedContent: ResponseData): OfficeSearch | null;
    static validateDocumentSearch(content: ResponseData): ResponseData | null;
    static normalizeDocumentSearch(untrustedContent: ResponseData): DocumentSearch | null;
    static validateKnowledgebaseSearch(content: ResponseData): ResponseData | null;
    static normalizeKnowledgebaseSearch(untrustedContent: ResponseData): KnowledgeSearch | null;
    static validateCatalogSearch(content: ResponseData): ResponseData | null;
    static normalizeCatalogSearch(untrustedContent: ResponseData): CatalogSearch | null;
    static normalizeOffice(untrustedContent: ResponseData): Office | null;
    static normalizeSearch(response: ResponseData): ResponseData;
    static normalizeSearchSuggestions(response: ResponseData): ResponseData;
    static normalizeNotification(untrustedContent: ResponseData): Notification | null;
    static normalizeManualNotification(untrustedContent: ResponseData): ManualNotification | null;
    static validateManualNotification(content: ResponseData): ResponseData | null;
    static dataToObjectManualNotification(contentData: ResponseData): ManualNotification;
    static normalizeManualNotificationFilterList(content: ResponseData): Categories[];
    static normalizeManualNotificationFilters(content: ResponseData): Categories[];
    static normalizeManualNotificationFilterData(untrustedContent: ResponseData): Categories | null;
    static validateManualNotificationFilterList(content: ResponseData): ResponseData | null;
    static normalizeArticles(response: ResponseData): ResponseData;
    static normalizeEvents(response: ResponseData): ResponseData;
    static normalizeFacets(responseFacets: ResponseData): ResponseData;
    static normalizeDepartments(response: ResponseData): Department[];
    static normalizeLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeQuickLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeTodo(untrustedContent: ResponseData): Todo | null;
    static normalizeBookmark(untrustedContent: ResponseData): Bookmark | null;
    static normalizeMenuLinks(content: ResponseData): LinkGroupItems[];
    static normalizeUserLinks(trustedContent: ResponseData): string[];
    static normalizeTodos(trustedContent: ResponseData): Todo[];
    static normalizeBookmarks(trustedContent: ResponseData): Bookmark[];
    static normalizePeople(content: ResponseData): PeopleModel[];
    static normalizeManualNotificationList(content: ResponseData): ManualNotificationListModel;
    static normalizeManualNotifications(content: ResponseData): ManualNotifications;
    static normalizeNotificationsBell(content: ResponseData): NotificationBell | null;
    static normalizeNotificationBell(untrustedContent: ResponseData): NotificationBell | null;
    static validateNotificationBell(content: ResponseData): ResponseData | null;
    static normalizeNotificationsArchive(content: ResponseData): boolean | null;
    static normalizeNotificationArchive(untrustedContent: ResponseData): boolean | null;
    static validateNotificationArchive(content: ResponseData): ResponseData | null;
    static normalizeArticleById(content: ResponseData): Article[];
    static normalizeEventById(content: ResponseData): EventModel[];
    static validateUser(content: ResponseData): ResponseData | null;
    static userAndProfileContent(untrustedContent: ResponseData): UserInfoModel | null;
    static userProfilePictureContent(untrustedContent: ResponseData): UserProfilePicture | null;
    static userProfileContent(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUser(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeProfile(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeUsers(content: ResponseData): UserInfoModel[];
    static normalizePage(response: ResponseData): Page | null;
    static normalizeUserProfile(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUserProfilePicture(untrustedContent: ResponseData): UserProfilePicture | null;
    static normalizeOrgChart(content: ResponseData): Users;
    static normalizeSubordinates(untrustedContent: ResponseData): Manager | null;
    static validateSubordinates(content: ResponseData): ResponseData | null;
    static normalizeOfficesPeoples(content: ResponseData): OfficesPeople[];
    static normalizeOfficesPeople(untrustedContent: ResponseData): OfficesPeople | null;
    static normalizeOfficeContents(content: ResponseData): OfficeItem[];
    static normalizeSetUserProfileData(content: any): any;
    static normalizeOfficeContent(content: ResponseData): OfficeItem | null;
    static validateOfficesPeople(content: ResponseData): ResponseData | null;
    static validateOfficeContent(content: ResponseData): ResponseData | null;
    static normalizeUserProfileUpdate(content: ResponseData): UserProfileUpdateResponse;
    static normalizeMyContacts(content: ResponseData): ContactList | null;
    static normalizeMyContactsResponse(content: ResponseData): ContactList[];
    static normalizeUsersByEmail(content: ResponseData): UserInfoModel[];
    static normalizeProfilepicDeleteResponse(content: ResponseData): ContactResponse | null;
}
//# sourceMappingURL=apiNormalizer.d.ts.map