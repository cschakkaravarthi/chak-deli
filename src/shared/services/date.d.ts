export declare const formatDate: (unixTimestamp?: number | undefined, dateTime?: string | undefined, format?: string) => string;
/**
 * Return Event Start Time if duration is 23:59:00 or less
 * hours * mints = 24 * 60 = 1440 - 1 mint = 1430
 * @param unixTimestampDuration
 * @param unixTimestampStartTime
 * @param format
 */
export declare const getEventTime: (unixTimestampDuration: number, unixTimestampStartTime: number, format?: string) => string;
export declare const formatDateTime: (unixTimestamp?: number | undefined, format?: string) => string;
export declare const formatDateRange: (startDate: number, endDate: number) => string;
export declare const formatDateRangeWithTime: (startDate: number, endDate: number) => string;
export declare const currentDate: (format?: string) => string;
interface ValidateDateErrorMessage {
    validDateError: string;
    bothFieldError: string;
}
export declare const validateDate: (day: string, month: string, message: ValidateDateErrorMessage) => string;
export {};
//# sourceMappingURL=date.d.ts.map