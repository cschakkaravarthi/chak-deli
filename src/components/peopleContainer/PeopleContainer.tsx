import React, { FC, useEffect, useState } from 'react';
import { Office as OfficeModel } from '../../shared/models/Office.model';
import { AlphabeticalLine } from 'umgc_ui_library/lib';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import PeopleModel from '../../shared/models/People.model';
import People from '../user/People';
import { handleScrollToBottom } from '../../utils/scroller';
import { PEOPLE_LIMIT, alphabet } from '../../constants/constants';
import Form from 'react-bootstrap/Form';
import images from '../../images/images';
import { OnError as IsEmpty } from '../error/Error';

type Props = {
  isActive?: string;
  isLastPage: boolean;
  office?: OfficeModel;
  people?: PeopleModel[];
  clearFilter?: () => void;
  handleActiveItem?: (letter: string) => void;
  triggerFetchPeople: <T>() => Promise<T> | undefined;
};

export const PeopleContainer: FC<Props> = props => {
  const {
    people,
    isActive,
    isLastPage,
    clearFilter,
    handleActiveItem,
    triggerFetchPeople
  } = props;

  const [isFetching, setIsFetching] = useState<boolean>(false);

  const fetchMorePeople = async (): Promise<any> => {
    await triggerFetchPeople();

    return setIsFetching(false);
  };

  useEffect(() => {
    if (isFetching) {
      return;
    }

    window.addEventListener('scroll', () =>
      isFetching || handleScrollToBottom(() => setIsFetching(true))
    );

    return () => {
      window.removeEventListener('scroll', () =>
        isFetching || handleScrollToBottom(() => setIsFetching(true))
      );
    };
  }, [isFetching]);

  useEffect(() => {
    if (!isFetching) {
      return;
    }

    if (people && people.length < PEOPLE_LIMIT) {
      setIsFetching(false);
      return;
    }

    if (!isLastPage) {
      fetchMorePeople();
    } else {
      setIsFetching(false);
    }
  }, [isFetching]);

  return (
    <>
      <p className="font-size-xl mb-3 mt-5">
        <b>People</b>
      </p>
      <Row>
        <Col>
          <Form className="d-flex d-md-none justify-content-start justify-content-md-end py-3">
            <Form.Label className="d-flex justify-content-start justify-content-md-end mr-3 mb-0 align-items-center">
              By letter:
            </Form.Label>
            <Form.Control
              size="sm"
              as="select"
              value={isActive}
              id="filter-alphabet"
              name="filter-alphabet"
              className="w-auto rounded-0"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                handleActiveItem && handleActiveItem(e.target.value)
              }
            >
              <option value={0}>Choose a letter</option>
              {alphabet.map((item: string) => (
                <option key={item} value={item}>
                  {item.toUpperCase()}
                </option>
              ))}
            </Form.Control>
          </Form>
          <div className="py-3 alphabetical-line d-none d-md-flex flex-wrap justify-content-md-between">
            {alphabet.map((item: string) => (
              <AlphabeticalLine
                className="pr-3 p-md-0"
                key={item.toUpperCase()}
                letter={item.toUpperCase()}
                isActive={item === isActive}
                onClick={() => handleActiveItem && handleActiveItem(item)}
              />
            ))}
          </div>
        </Col>
      </Row>
      {isActive && people ? (
        <Row>
          <Col className="d-flex just-content-start align-items-baseline">
            <p className="font-size-sm">Filtered by:</p>
            <button onClick={clearFilter} className="ml-2 mr-1">
              <img src={images.remove} alt="clean filter" width="12" height="12" />
            </button>
            <p className="font-size-sm">Letter: <strong> {isActive.toUpperCase()}</strong></p>
          </Col>
        </Row>
      ) : null}
      <Row className="row-cols-lg-5">
        {people &&
          people.map((person: PeopleModel) => (
            <People people={person} key={person.loginId} />
          ))}
      </Row>
      <Row>
        <Col>
          {!!isLastPage && people && !people.length && <IsEmpty />}
          {!!isFetching && !isLastPage && !!people && !!people.length && (
            <Col className="p-5 d-flex justify-content-center text-secondary">
              <strong>Fetching people ...</strong>
            </Col>
          )}
        </Col>
      </Row>
    </>
  );
};

export default PeopleContainer;
