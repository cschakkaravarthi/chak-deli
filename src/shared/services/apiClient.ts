import axios from 'axios';
import get from 'lodash.get';
import packageJson from '../../../package.json';
import { getAccessToken } from '../../reactAuthProvider';
import { appEnv } from '../../utils/customHooks';

const apiVersion = packageJson.umgcApi.useEdge ? 'dev/' : `v${packageJson.umgcApi.version}/`;

// For some reason appEnv is undefined at this point, but window.env is not?
const client = axios.create({
  baseURL: get(appEnv || window.env, 'centralApi.baseUrl') + apiVersion
});

client.interceptors.request.use(request => {
  return new Promise((resolve, reject) => {
    getAccessToken()
      .then(accessTokenRepsonse => {
        request.headers.Authorization = 'Bearer ' + accessTokenRepsonse;
        resolve(request);
      })
      .catch(() => {
        // console.log('the catch token is ' + client.defaults.headers.common.Authorization);
        reject(request);
      });
  });
});

// @TODO: Logs for dev only.
client.interceptors.request.use(request => {
  // console.log('Starting Request:');
  // console.log(request);
  return request;
});

client.interceptors.response.use(response => {
  // console.log('Response:');
  // console.log(response);
  return response;
});

export default client;
