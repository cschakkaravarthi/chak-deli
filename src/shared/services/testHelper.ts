import { ReactElement } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import getStore from './mockGlobalStore';

export const shallowRender = (toRender: JSX.Element): ReactElement => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};

export const getMockStore = getStore;

export const drawerMockNavigation = {
  state: {
    isDrawerOpen: true,
    isTransitioning: false,
    index: 1,
    routes: [],
    key: 'abc',
    params: {}
  },
  dispatch: jest.fn(),
  goBack: jest.fn(),
  dismiss: jest.fn(),
  navigate: jest.fn(),
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  dangerouslyGetParent: jest.fn()
};
