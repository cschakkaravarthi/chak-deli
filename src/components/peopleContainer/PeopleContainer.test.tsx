import React from 'react';
import PeopleContainer from './PeopleContainer';
import { shallowRender } from '../../shared/services/testHelper';

describe('PeopleContainer', () => {
  it('renders null', () => {
    const tree = shallowRender(<PeopleContainer
      people={[]}
      isLastPage={false}
      triggerFetchPeople={jest.fn()}/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders peopleContainer', () => {
    const tree = shallowRender(<PeopleContainer
      isLastPage={false}
      triggerFetchPeople={jest.fn()}/>);
    expect(tree).toMatchSnapshot();
  });
});
