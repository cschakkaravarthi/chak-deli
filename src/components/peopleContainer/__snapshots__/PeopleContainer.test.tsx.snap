// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleContainer renders null 1`] = `
<React.Fragment>
  <p
    className="font-size-xl mb-3 mt-5"
  >
    <b>
      People
    </b>
  </p>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef>
      <ForwardRef
        className="d-flex d-md-none justify-content-start justify-content-md-end py-3"
        inline={false}
      >
        <ForwardRef
          className="d-flex justify-content-start justify-content-md-end mr-3 mb-0 align-items-center"
          column={false}
          srOnly={false}
        >
          By letter:
        </ForwardRef>
        <ForwardRef
          as="select"
          className="w-auto rounded-0"
          id="filter-alphabet"
          name="filter-alphabet"
          onChange={[Function]}
          size="sm"
        >
          <option
            value={0}
          >
            Choose a letter
          </option>
          <option
            value="a"
          >
            A
          </option>
          <option
            value="b"
          >
            B
          </option>
          <option
            value="c"
          >
            C
          </option>
          <option
            value="d"
          >
            D
          </option>
          <option
            value="e"
          >
            E
          </option>
          <option
            value="f"
          >
            F
          </option>
          <option
            value="g"
          >
            G
          </option>
          <option
            value="h"
          >
            H
          </option>
          <option
            value="i"
          >
            I
          </option>
          <option
            value="j"
          >
            J
          </option>
          <option
            value="k"
          >
            K
          </option>
          <option
            value="l"
          >
            L
          </option>
          <option
            value="m"
          >
            M
          </option>
          <option
            value="n"
          >
            N
          </option>
          <option
            value="o"
          >
            O
          </option>
          <option
            value="p"
          >
            P
          </option>
          <option
            value="q"
          >
            Q
          </option>
          <option
            value="r"
          >
            R
          </option>
          <option
            value="s"
          >
            S
          </option>
          <option
            value="t"
          >
            T
          </option>
          <option
            value="u"
          >
            U
          </option>
          <option
            value="v"
          >
            V
          </option>
          <option
            value="w"
          >
            W
          </option>
          <option
            value="x"
          >
            X
          </option>
          <option
            value="y"
          >
            Y
          </option>
          <option
            value="z"
          >
            Z
          </option>
        </ForwardRef>
      </ForwardRef>
      <div
        className="py-3 alphabetical-line d-none d-md-flex flex-wrap justify-content-md-between"
      >
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="A"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="B"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="C"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="D"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="E"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="F"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="G"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="H"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="I"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="J"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="K"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="L"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="M"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="N"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="O"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="P"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Q"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="R"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="S"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="T"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="U"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="V"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="W"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="X"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Y"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Z"
          onClick={[Function]}
        />
      </div>
    </ForwardRef>
  </ForwardRef>
  <ForwardRef
    className="row-cols-lg-5"
    noGutters={false}
  />
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef />
  </ForwardRef>
</React.Fragment>
`;

exports[`PeopleContainer renders peopleContainer 1`] = `
<React.Fragment>
  <p
    className="font-size-xl mb-3 mt-5"
  >
    <b>
      People
    </b>
  </p>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef>
      <ForwardRef
        className="d-flex d-md-none justify-content-start justify-content-md-end py-3"
        inline={false}
      >
        <ForwardRef
          className="d-flex justify-content-start justify-content-md-end mr-3 mb-0 align-items-center"
          column={false}
          srOnly={false}
        >
          By letter:
        </ForwardRef>
        <ForwardRef
          as="select"
          className="w-auto rounded-0"
          id="filter-alphabet"
          name="filter-alphabet"
          onChange={[Function]}
          size="sm"
        >
          <option
            value={0}
          >
            Choose a letter
          </option>
          <option
            value="a"
          >
            A
          </option>
          <option
            value="b"
          >
            B
          </option>
          <option
            value="c"
          >
            C
          </option>
          <option
            value="d"
          >
            D
          </option>
          <option
            value="e"
          >
            E
          </option>
          <option
            value="f"
          >
            F
          </option>
          <option
            value="g"
          >
            G
          </option>
          <option
            value="h"
          >
            H
          </option>
          <option
            value="i"
          >
            I
          </option>
          <option
            value="j"
          >
            J
          </option>
          <option
            value="k"
          >
            K
          </option>
          <option
            value="l"
          >
            L
          </option>
          <option
            value="m"
          >
            M
          </option>
          <option
            value="n"
          >
            N
          </option>
          <option
            value="o"
          >
            O
          </option>
          <option
            value="p"
          >
            P
          </option>
          <option
            value="q"
          >
            Q
          </option>
          <option
            value="r"
          >
            R
          </option>
          <option
            value="s"
          >
            S
          </option>
          <option
            value="t"
          >
            T
          </option>
          <option
            value="u"
          >
            U
          </option>
          <option
            value="v"
          >
            V
          </option>
          <option
            value="w"
          >
            W
          </option>
          <option
            value="x"
          >
            X
          </option>
          <option
            value="y"
          >
            Y
          </option>
          <option
            value="z"
          >
            Z
          </option>
        </ForwardRef>
      </ForwardRef>
      <div
        className="py-3 alphabetical-line d-none d-md-flex flex-wrap justify-content-md-between"
      >
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="A"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="B"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="C"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="D"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="E"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="F"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="G"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="H"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="I"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="J"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="K"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="L"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="M"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="N"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="O"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="P"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Q"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="R"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="S"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="T"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="U"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="V"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="W"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="X"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Y"
          onClick={[Function]}
        />
        <Unknown
          className="pr-3 p-md-0"
          isActive={false}
          letter="Z"
          onClick={[Function]}
        />
      </div>
    </ForwardRef>
  </ForwardRef>
  <ForwardRef
    className="row-cols-lg-5"
    noGutters={false}
  />
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef />
  </ForwardRef>
</React.Fragment>
`;
